### ISSUE TITLE
Dog Cages explained (+issues)

### ISSUE THREAD

**Issue by Truetobi91** [Sh0wdown/UnofficialCrusaderPatch#681]
So I did some testing with Dog Cages yesterday as I didn't exactly know how they work and I bet most players don't know that too.

Dog cages cost 10 wood and 100 gold and require a space of 3x3 tiles. They detect enemy numbers within a range of 16 tiles as shown in the picture below. Once the enemy numbers reach 6 units the dogs emerge from their cage and attack nearby enemies. Dogs will also hunt their currently locked target beyond the detection radius. Dogs can split up to also attack enemies that are currently not locked by another dog but will not always do so right from the start. Dogs fight of any remaining enemies in the area and return to their cage once the area is clear.

![grafik](https://user-images.githubusercontent.com/48789759/98099027-5a5af880-1e8f-11eb-8c25-bcb9fd262232.png)

Now the dogs aren't the best fighters around when fully stacked they can kill 6 EuropArchers or 6 ArabArchers in melee combat but will die to 6 Slingers or Spearmen. They can also kill 9 slaves or even a bit more depending on stacking and focus. 

Remaining dogs will return to the cage but will not replenish so even though the cage shows the same animation its protective capabilities decrease by fighting. Once all dogs are killed the cage is automatically destroyed. 

Issues: 
- Groups of 5 units e.g. slaves can destroy a dog cage without the dogs being triggered to attack. 
- Dogs sometimes run back to their cage when the enemy unit count in the area drops below 6 while some units remain. 

Personal notes:
- Dog Cages should trigger earlier with a unit count of 1. 
- Price should be reduced to just gold
- Dogs could have their toughness increased or not-instant replenishment enabled.

**Comment by ilosevsrat** [comment]
i think Dogs are fine, after all its just a dog...how much damage it should deal to armoured  human?

but small buff is needed for sure,  i strongly believe That when dogs return in cage they should replenish Hp and  their numbers.

**Comment by GRhin** [comment]
For you general fund of knowledge, these dogs have the health/resistance stats of slaves, but do the damage of a monk - thats 2.5 x the damage of a spear or pike, more than any ranged unit trying to melee, and 5x a slave. for essentialy 20gold+4wood each, i think thats fair. I definately dont think that the replenishment should be enabled, that would make them too strong.
The wood cost i dont think is a big deal imo, i mean, you are talking about a 4 gold difference per dog. But I dont feel strongly either way.
The other 2 points are definately ones to consider.
Curious about your first image, did you test with units in the corners? I know that the aggressive range is roughly circular, as is the archer range

**Comment by GRhin** [comment]
I would suggest, if the trigger number is reduced to 1, that you reduce the trigger range rather  than increase hp, this will mean they take less ranged damage, so are more effective in a melee fight, but not make them too locally strong. I see them being used as a small raid defender in that case, stopping slaves and spears allowing other troops to deal with the stronger and ranged units.

**Comment by Truetobi91** [comment]
> 
> 
> For you general fund of knowledge, these dogs have the health/resistance stats of slaves (edit: not quite true, they tend to take 2x damage from melee as well, while the dogs dont. Maybe 60% of a spear is a better example), but do the damage of a monk - thats 2.5 x the damage of a spear or pike, more than any ranged unit trying to melee, and 5x a slave. for essentialy 20gold+4wood each, i think thats fair. I definately dont think that the replenishment should be enabled, that would make them too strong.
> The wood cost i dont think is a big deal imo, i mean, you are talking about a 4 gold difference per dog. But I dont feel strongly either way.
> The other 2 points are definately ones to consider.
> Curious about your first image, did you test with units in the corners? I know that the aggressive range is roughly circular, as is the archer range

Yes, I put 5 slaves right next to the dog cage and let one other slave approach the dog cage from different angles

**Comment by Heroesflorian** [comment]
@Truetobi91 dog cage triggering mechanics is not based on troop count alone

They trigger for a single european swordsman placed within range, for example: 
![grafik](https://user-images.githubusercontent.com/12549394/99891165-44a74a80-2c67-11eb-907c-d27db065a81e.png)

For weak troops like slaves, it will indeed take a higher amount, while for stronger units it takes a lower amount to trigger the dogs. Similar behaviour can, by the way, be observed for oil pot engineers who will trigger based on (their own stance and) the strength and amount of approaching enemy troops.

**Comment by Krarilotus** [comment]
i believe the trigger is thes same as for oil engineers or for lighting pitch, as in the internal strength value of troops, that is also calculated for the greatest lord score as impact on the army, and calculated for the lord to come out when defending against left over army (slingers for example count as 2, so it would need 3 slingers i suppse ?!)
